testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -241,6 +270,90 @@ impl BlockMinerThread {
         }
     }
 
+    fn wait_for_signed_block(
```

Comment:
Architecturally speaking, wouldn't it make more sense to have the node react to push notifications from arriving StackerDB chunks, instead of polling every so often?

Reply:
I am okay with reacting to push notifications, but this will be a more involved change as I think I will need to do some revamping to prevent having to spin a listener as a node to itself. (so I can directly access the callbacks.) Following convos with Aaron, he suggested this approach for first pass. But don't mind making the change now.